src/fiatExchanges/SelectProvider.tsx

```diff
@@ -0,0 +1,516 @@
+import React, { useState } from 'react'
+import {
+  LayoutAnimation,
+  StyleSheet,
+  Text,
+  View,
+  Image,
+  ActivityIndicator,
+  ScrollView,
+} from 'react-native'
+import Touchable from 'src/components/Touchable'
+import BackButton from 'src/components/BackButton'
+import Expandable from 'src/components/Expandable'
+import { emptyHeader } from 'src/navigator/Headers'
+import colors from 'src/styles/colors'
+import fontStyles from 'src/styles/fonts'
+import {
+  fetchProviders,
+  ProviderQuote,
+  sortQuotes,
+  isSimplexQuote,
+  getQuotes,
+  CicoQuote,
+  getFeeValueFromQuotes,
+  SimplexQuote,
+  PaymentMethod,
+  fetchLocalCicoProviders,
+  getAvailableLocalProviders,
+  LocalCicoProvider,
+  CICOFlow,
+} from './utils'
+import { StackScreenProps } from '@react-navigation/stack'
+import { StackParamList } from 'src/navigator/types'
+import { Screens } from 'src/navigator/Screens'
+import { useDispatch, useSelector } from 'react-redux'
+import { userLocationDataSelector } from 'src/networkInfo/selectors'
+import { currentAccountSelector } from 'src/web3/selectors'
+import { getLocalCurrencyCode } from 'src/localCurrency/selectors'
+import { CiCoCurrency, Currency } from 'src/utils/currencies'
+import { useAsync } from 'react-async-hook'
+import Logger from 'src/utils/Logger'
+import { showError } from 'src/alert/actions'
+import variables from 'src/styles/variables'
+import { ErrorMessages } from 'src/app/ErrorMessages'
+import CurrencyDisplay from 'src/components/CurrencyDisplay'
+import { useTranslation } from 'react-i18next'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import { FiatExchangeEvents } from 'src/analytics/Events'
+import { navigate } from 'src/navigator/NavigationService'
+import { navigateToURI } from 'src/utils/linking'
+import Dialog from 'src/components/Dialog'
+import i18n from 'src/i18n'
+import { RouteProp } from '@react-navigation/native'
+
+const TAG = 'SelectProviderScreen'
+
+type Props = StackScreenProps<StackParamList, Screens.SelectProvider>
+
+export default function SelectProviderScreen({ route, navigation }: Props) {
+  const dispatch = useDispatch()
+  const userLocation = useSelector(userLocationDataSelector)
+  const account = useSelector(currentAccountSelector)
+  const localCurrency = useSelector(getLocalCurrencyCode)
+  const [noPaymentMethods, setNoPaymentMethods] = useState(false)
+  const { flow } = route.params
+
+  const currencyToBuy = {
```

Comment:
i thought this const also is used in withdraw flow, it's a bit confusing to name it `currencyToBuy`, maybe `currencyInTransaction`